jax/_src/extend/ffi.py

```diff
@@ -27,7 +27,7 @@
 from jax._src import dispatch
 from jax._src import effects
 from jax._src import util
-from jax._src.callback import _check_shape_dtype, callback_batching_rule
```

Comment:
Does it make sense to do the same change in callbacks? I'm pretty sure you can return a I64 from it, assuming you aren't using JAX in the body.

Reply:
Good question! That was added by @sharadmv in https://github.com/jax-ml/jax/pull/12186, 
(I expect that was long enough ago that he doesn't remember the context :D) but I agree that this case probably should be allowed when explicitly requested. There probably was a good reason to add this in the first place, and it seems like sensible behavior to catch cases where this would be a mistake. One option would be to put it behind a flag like `allow_x64` so that we still error in cases where a user doesn't explicitly promise that they know what they're doing. What do you think?

Reply:
Let's ask Sharad separately and land this! :)

Reply:
After offline discussions, we decided to remove the canonicalization check from `pure_callback` and `io_callback` too.

Reply:
Scratch that! It looks like canonicalization is deeply baked into the callback implementation, so I'm not going to touch that here. If someone needs to be able to return `x64` from a callback in the future, let's cross that bridge when we get there :D